src/index.ts

```diff
@@ -139,6 +139,7 @@ const exchangeSchema = executableExchangeSchema(legacyTransformsForExchange)
 
 const graphqlHTTP = require("express-graphql")
 const graphqlServer = graphqlHTTP((req, res, params) => {
+  const requestIP = req.ip as string | undefined
```

Comment:
Is this giving us the IP of the request and can we use this IP for the geolocation?

Reply:
Not necessarily. You may need to pull out the `x-forwarded-for` header. It seems like it'd be safer to accept IP as an argument like `requestLocation` does. Alternatively, I could see us exposing a `localized` type instead of `requestLocation` that has things like shows nested under it, though that might be getting into the weeds.

Reply:
Yea, MP is already an extra 'hop' away from the actual user's request. Force might be a more natural place to pull that out (I would think Force might already try to pull that out in some middleware or make it available somewhere).

Generally if `x-forwarded-for` is set (as a CSV string), the _first_ IP in the chain should correspond with the original requester. If it is blank or not set, `req.connection.remoteAddress` (some Express property) will be more accurate. It's specific to how MP is being ran (behind nginx, locally in dev, etc.) so it's good to support all scenarios.

Reply:
> Force might be a more natural place to pull that out 

That sounds like a good idea. I've added a new parameter, `ip`, and removed `includeShowsByLocation`. The params`near` and `ip` are not required (in this case local shows are not added) and mutually exclusive. 

We plan to check in Eigen to see if the user has given permission to access the location. If not, we'll fall back to the IP for now. The rail will be displayed on the home screen, so it would be intrusive if the permission dialog just appeared after the app was opened.

Reply:
Coming back to this discussion. `showsConnection` currently takes IP as an argument, so the client needs to determine the IP address. This has led us to make another API call in Eigen to get the IP, which isn't ideal (https://github.com/artsy/eigen/pull/8758).

I am thinking about changing the implementation so that we use the IP address from the request, rather than expecting it as an argument (making the IP argument optional).

I'm not certain how to test this, but would this consistently work to get the client's IP address?

```
req.headers["x-forwarded-for"] || req.connection.remoteAddress
```

Reply:
Something like that, except `x-forwarded-for` can be blank, contain 1 IP address, _or_ a comma-delimited list of IP addresses like @mzikherman mentions. In that case, the _first_ IP is the "original."